### PR TITLE
Add a ready attribute to await Helm Charts

### PR DESCRIPTION
**Pull request by lblackstone**
### Proposed changes

After chatting with @pgavlin about this problem, we determined that Helm Charts were not awaiting properly because child resources are not known at the time the Chart is initialized. The most promising solution appears to be exposing a field on the Chart class that can be passed into the `dependsOn` option. The existing `resources` field is a map rather than an array, so we either need to expand `dependsOn` support to include maps, or create a new field on the Chart class that works with the existing definition.

This PR takes the latter approach, and creates a new `.ready` field that can be passed directly to the `dependsOn` option. This change makes dependent resources wait until all Chart subresources are ready as expected. Here's an example program to demonstrate:

```typescript
import * as k8s from "@pulumi/kubernetes";

const ns = new k8s.core.v1.Namespace("foo")
const chart = new k8s.helm.v3.Chart("nginx", {
    chart: "nginx",
    namespace: ns.metadata.name,
    values: {
        service: {
            type: "ClusterIP"
        }
    },
    fetchOpts: {
        repo: "https://charts.bitnami.com/bitnami"
    },
})

new k8s.core.v1.ConfigMap("test", {
    data: {foo: "bar"}
}, {dependsOn: chart.ready})
```

Here's what this looks like before/after the change (notice the `test` `ConfigMap` gets created immediately on `Before`):
## Before
Before

## After
After

### Related issues (optional)
Related to #861

### CODE REVIEW

**Review by pgavlin — COMMENTED** [review]
This looks good to me. Any ideas on how to test this?

**Comment by lblackstone** [comment]
> Hey @lblackstone what would the code look like for the python sdk, I am guessing very similar. Wanted to give it a try.

I haven't had a chance to work on it yet, but should be the same basic concept. Easiest way to test out would be modifying the relevant Helm/YAML SDK files in your virtualenv

**Comment by lblackstone** [comment]
> Any ideas on how to test this?

I'm not sure yet. I'd like to verify that the ConfigMap isn't created until after all of the Chart resources, but I'm not sure how to get that information. I don't suppose there's an easy way to get timestamps of resource creation events? @pgavlin

**Comment by lblackstone** [comment]
Note: this PR is only for NodeJS. Once this merges, I'll follow with the other SDKs

**Review by komalali — APPROVED** [review]

File: provider/pkg/gen/nodejs-templates/helm/v2/helm.ts [hunk 1]
```diff
@@ -254,6 +254,8 @@ export class Chart extends yaml.CollectionComponentResource {
                 overrides.removeCallback();
             }
         });
+
+        this.ready = this.resources.apply(m => pulumi.all(m).apply(m => Object.values(m).map(r => pulumi.output(r))));
```
[the remark this answers is not in the document]
Comment (lblackstone) [reply provider/pkg/gen/nodejs-templates/helm/v2/helm.ts]: It does indeed. Thanks for the suggestion!

File: provider/pkg/gen/nodejs-templates/helm/v2/helm.ts [hunk 2]
```diff
@@ -255,7 +255,7 @@ export class Chart extends yaml.CollectionComponentResource {
             }
         });
 
-        this.ready = this.resources.apply(m => Object.values(m).map(r => pulumi.output(r)));
+        this.ready = this.resources.apply(m => Object.values(m));
```
Comment (komalali) [inline provider/pkg/gen/nodejs-templates/helm/v2/helm.ts:258]: Ah nice, I was just gonna say the same thing ✅

**Pull request merged**